subcmd/subcmd.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Package subcmd implements simple command line interface subcommands.
+package subcmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+// ParseFunc parses all flags that have been set up. Include extra handling for "-h" help flag
+// detailed output, invalid args, and error conditions.
+type ParseFunc func() error
+
+// Option implements one subcommand option that the user may pick.
+type Option interface {
```

Comment:
As discussed offline, is not modeling behavior but just enumerating properties. The only behavior here are `Handle` and `ArgsSummary`.

Therefore, it would be better to define it as a struct which contains good old props and two functions pointers.